### ISSUE TITLE
Error FromGroupState property undefined with validation required

### ISSUE THREAD

**Issue by romankhrystynych** [MrWolfZ/ngrx-forms#64]
There are errors when creating a FromGroupState with undefined property as initial state and validation.

```js
const FORM_ID = 'profile';

export interface IProfile {
  name: string;
  age: number;
}

export const INITIAL_STATE = createFormGroupState<IProfile>(FORM_ID, {
  name: undefined,
  age: undefined,
});

const validationFormGroupReducer = createFormGroupReducerWithUpdate<IProfile>({
  name: validate<string>(required),
});

export function profileReducer(_s: any, _a: any) {
  return combineReducers<any, any>({
    formState(s = INITIAL_STATE, a: Action) {
      return validationFormGroupReducer(s, a);
    },
  })(_s, _a);
}
```

Errors in browser bellow
```Error
TypeError: Cannot read property 'length' of undefined
    at required (validation.es5.js:434)
    at eval (forms.es5.js:2274)
    at Array.reduce (<anonymous>)
    at validate (forms.es5.js:2274)
    at Object.eval [as name] (forms.es5.js:2277)
    at eval (forms.es5.js:2076)
    at Array.reduce (<anonymous>)
    at eval (forms.es5.js:2072)
    at eval (forms.es5.js:2092)
    at eval (forms.es5.js:2114)

ProfileComponent.html:9 ERROR TypeError: Cannot read property 'profile' of undefined
    at MapSubscriber.exports.profileSelector [as project] (profile.selector.ts:4)
    at MapSubscriber._next (map.js:79)
    at MapSubscriber.Subscriber.next (Subscriber.js:95)
    at MapSubscriber._next (map.js:85)
    at MapSubscriber.Subscriber.next (Subscriber.js:95)
    at ReplaySubject._subscribe (ReplaySubject.js:61)
    at ReplaySubject.Observable._trySubscribe (Observable.js:172)
    at ReplaySubject.Subject._trySubscribe (Subject.js:98)
    at ReplaySubject.Observable.subscribe (Observable.js:160)
    at Observable._subscribe (Observable.js:231) 

ProfileComponent.html:9 ERROR TypeError: Cannot read property 'name' of null
    at Object.eval [as updateDirectives] (ProfileComponent.html:11)
    at Object.debugUpdateDirectives [as updateDirectives] (core.js:14655)
    at checkAndUpdateView (core.js:13802)
    at callViewAction (core.js:14153)
    at execComponentViewsAction (core.js:14085)
    at checkAndUpdateView (core.js:13808)
    at callViewAction (core.js:14153)
    at execEmbeddedViewsAction (core.js:14111)
    at checkAndUpdateView (core.js:13803)
    at callViewAction (core.js:14153)
```

No problem when value of name is set to null.

**Comment by MrWolfZ** [comment]
Indeed, the validation functions are only considering null since that is what browsers report from input elements etc. However, that is unintuitive and not well documented. I will fix this in the next version. For now changing the initial values to null should work.

**Comment by MrWolfZ** [comment]
I have just released version [3.0.0](https://github.com/MrWolfZ/ngrx-forms/releases/tag/v3.0.0) which supports `null` and `undefined` form control values.
